test/WebJobs.Script.Tests/Metrics/FlexConsumptionMetricsPublisherTests.cs

```diff
@@ -246,13 +246,196 @@ public void FunctionsStartStop_CorrectCountsAreMaintained()
             Assert.Equal(2, publisher.FunctionExecutionCount);
             Assert.Equal(0, publisher.FunctionExecutionTimeMS);
 
-            publisher.OnFunctionCompleted("foo", "111");
+            publisher.OnFunctionCompleted("baz", "333");
 
             Assert.Equal(0, publisher.ActiveFunctionCount);
             Assert.Equal(3, publisher.FunctionExecutionCount);
             Assert.True(publisher.FunctionExecutionTimeMS > 0);
         }
 
+        [Fact]
```

Comment:
Following are 3 scenario tests that test a bunch of different invocation patterns. Please follow through (maybe with paper and pencil :)) and verify intervals are being measured as intended.

Reply:
I verified with some scratch work that the test cases make sense!

Reply:
I followed these 3 scenarios through step by step. They make sense to me too.